### PR TITLE
Ad french translation

### PR DESCRIPTION
**Pull request by ColonelMoutarde**

### CODE REVIEW

File: platformio/src/locales/locale_fr_FR.inc [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+const char *LC_ERA_T_FMT   = "";
+
+// OWM LANGUAGE
+// For full list of languages, see 
+// https://openweathermap.org/api/one-call-api#multi
+// Note: "[only] The contents of the 'description' field will be translated."
+const String OWM_LANG = "fr";
+
+// CURRENT CONDITIONS
+const char *TXT_FEELS_LIKE         = "Feels Like";
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:73]: Missed this one?

File: platformio/src/locales/locale_fr_FR.inc [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+const char *TXT_SUNRISE            = "Lever de soleil";
+const char *TXT_SUNSET             = "Coucher de soleil";
+const char *TXT_WIND               = "Vent";
+const char *TXT_HUMIDITY           = "Humidité";
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:77]:
```suggestion
const char *TXT_HUMIDITY           = "Humidit\xE9";
```

File: platformio/src/locales/locale_fr_FR.inc [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+const char *TXT_UV_INDEX           = "Indice UV";
+const char *TXT_PRESSURE           = "Pression";
+const char *TXT_AIR_QUALITY_INDEX  = "Qualité de l'air";
+const char *TXT_VISIBILITY         = "Visibilité";
+const char *TXT_INDOOR_TEMPERATURE = "Temperature";
+const char *TXT_INDOOR_HUMIDITY    = "Humidité";
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:83]:
```suggestion
const char *TXT_AIR_QUALITY_INDEX  = "Qualit\xE9"" de l'air";
const char *TXT_VISIBILITY         = "Visibilit\xE9";
const char *TXT_INDOOR_TEMPERATURE = "Temp\xE9""rature";
const char *TXT_INDOOR_HUMIDITY    = "Humidit\xE9";
```

File: platformio/src/locales/locale_fr_FR.inc [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+
+// UV INDEX
+const char *TXT_UV_LOW       = "Bas";
+const char *TXT_UV_MODERATE  = "Moderaté";
+const char *TXT_UV_HIGH      = "Haut";
+const char *TXT_UV_VERY_HIGH = "Très haut";
+const char *TXT_UV_EXTREME   = "Extreme";
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:90]:
```suggestion
const char *TXT_UV_MODERATE  = "Moderat\xE9";
const char *TXT_UV_HIGH      = "Haut";
const char *TXT_UV_VERY_HIGH = "Tr\xE8""s haut";
const char *TXT_UV_EXTREME   = "Extr\xEA""me";
```

File: platformio/src/locales/locale_fr_FR.inc [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+
+// WIFI
+const char *TXT_WIFI_EXCELLENT     = "Excellent";
+const char *TXT_WIFI_GOOD          = "Bon";
+const char *TXT_WIFI_FAIR          = "juste";
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:95]: Fix inconsistent capitalization.
```suggestion
const char *TXT_WIFI_FAIR          = "Juste";
```

File: platformio/src/locales/locale_fr_FR.inc [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+const std::vector<String> TERM_STORM_WARNING =
+    {"storm warning"};
+const std::vector<String> TERM_HURRICANE_WARNING =
+    {"vent de type ouragan", "vent extreme", "grand vent"};
+const std::vector<String> TERM_HURRICANE =
+    {"ouragan", "tempete tropicale", "typhoon", "cyclone"};
+const std::vector<String> TERM_DUST =
+    {"poussière", "sable"};
+const std::vector<String> TERM_STRONG_WIND =
+    {"vent"};
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:203]: > //   OpenWeatherMap returns alerts in English regardless of the OWM LANGUAGE
> //   option or territory. For this reason it is preferred that you do not
> //   translate text related to Alerts.

This text is not shown to the user, just used to display the correct icon. As mentioned above, OpenWeatherMap only provides alerts in English. 

```suggestion
const std::vector<String> TERM_SMOG =
    {"smog"};
const std::vector<String> TERM_SMOKE =
    {"smoke"};
const std::vector<String> TERM_FOG =
    {"fog", "haar"};
const std::vector<String> TERM_METEOR =
    {"meteor", "asteroid"};
const std::vector<String> TERM_NUCLEAR =
    {"nuclear", "ionizing radiation"};
const std::vector<String> TERM_BIOHAZARD =
    {"biohazard", "biological hazard"};
const std::vector<String> TERM_EARTHQUAKE =
    {"earthquake"};
const std::vector<String> TERM_TSUNAMI =
    {"tsunami"};
const std::vector<String> TERM_FIRE =
    {"fire", "red flag"};
const std::vector<String> TERM_HEAT =
    {"heat"};
const std::vector<String> TERM_WINTER =
    {"blizzard", "winter", "ice", "snow", "sleet", "cold", "freezing rain", 
     "wind chill", "freeze", "frost", "hail"};
const std::vector<String> TERM_LIGHTNING =
    {"thunderstorm", "storm cell", "pulse storm", "squall line", "supercell",
     "lightning"};
const std::vector<String> TERM_SANDSTORM =
    {"sandstorm", "blowing dust", "dust storm"};
const std::vector<String> TERM_FLOOD =
    {"flood", "storm surge", "seiche", "swell", "high seas", "high tides",
     "tidal surge"};
const std::vector<String> TERM_VOLCANO =
    {"volcanic", "ash", "volcano", "eruption"};
const std::vector<String> TERM_AIR_QUALITY =
    {"air", "stagnation", "pollution"};
const std::vector<String> TERM_TORNADO =
    {"tornado"};
const std::vector<String> TERM_SMALL_CRAFT_ADVISORY =
    {"small craft", "wind advisory"};
const std::vector<String> TERM_GALE_WARNING =
    {"gale"};
const std::vector<String> TERM_STORM_WARNING =
    {"storm warning"};
const std::vector<String> TERM_HURRICANE_WARNING =
    {"hurricane force wind", "extreme wind", "high wind"};
const std::vector<String> TERM_HURRICANE =
    {"hurricane", "tropical storm", "typhoon", "cyclone"};
const std::vector<String> TERM_DUST =
    {"dust", "sand"};
const std::vector<String> TERM_STRONG_WIND =
    {"wind"};
```

File: platformio/src/locales/locale_fr_FR.inc [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+const char *UNITED_STATES_AQI_TXT[6] =
+{
+  "Bon",
+  "Modéré",
+  "Malsain pour les groupes sensibles",
+  "Malsain",
+  "Très malsain",
+  "Dangereux",
+};
+} // end extern "C"
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:288]:
```suggestion
// AIR QUALITY INDEX
extern "C" {
const char *AUSTRALIA_AQI_TXT[6] =
{
  "Tr\xE8""s bon",
  "Bon",
  "Juste",
  "Mauvais",
  "Tr\xE8""s mauvais",
  "Dangereux",
};
const char *CANADA_AQHI_TXT[4] =
{
  "Bas",
  "Mod\xE9""r\xE9",
  "Haut",
  "Tr\xE8""s haut",
};
const char *EUROPE_CAQI_TXT[5] =
{
  "Tr\xE8""s bas",
  "Bas",
  "Moyen",
  "Haut",
  "Tr\xE8""s haut",
};
const char *HONG_KONG_AQHI_TXT[5] =
{
  "Bas",
  "Mod\xE9""r\xE9",
  "Haut",
  "Tr\xE8""s haut",
  "Dangereux",
};
const char *INDIA_AQI_TXT[6] =
{
  "Bon",
  "Satisfaisant",
  "Mod\xE9""r\xE9",
  "Mauvais",
  "Tr\xE8""s mauvais",
  "S\xE9""v\xE8""re",
};
const char *MAINLAND_CHINA_AQI_TXT[6] =
{
  "Excellent",
  "Bon",
  "L\xE9""g\xE8""rement pollu\xE9",
  "Mod\xE9""r\xE9""ment pollu\xE9",
  "Tr\xE8""s pollu\xE9",
  "S\xE9""v\xE8""rement pollu\xE9",
};
const char *SINGAPORE_PSI_TXT[5] =
{
  "Bon",
  "Mod\xE9""r\xE9",
  "Malsain",
  "Tr\xE8""s malsain",
  "Dangereux",
};
const char *SOUTH_KOREA_CAI_TXT[4] =
{
  "Bon",
  "Moyen",
  "Malsain",
  "Tr\xE8""s malsain",
};
const char *UNITED_KINGDOM_DAQI_TXT[4] =
{
  "Bas",
  "Mod\xE9""r\xE9",
  "Haut",
  "Tr\xE8""s haut",
};
const char *UNITED_STATES_AQI_TXT[6] =
{
  "Bon",
  "Mod\xE9""r\xE9",
  "Malsain pour les groupes sensibles",
  "Malsain",
  "Tr\xE8""s malsain",
  "Dangereux",
};
} // end extern "C"
```

**Review by lmarzen — CHANGES_REQUESTED** [review]
Please address my comments.

File: platformio/src/locales/locale_fr_FR.inc (hunk 1 above)
Comment (ColonelMoutarde) [reply platformio/src/locales/locale_fr_FR.inc:73]: hi, to translate this term, i need some context, i can't quite figure out what it refers to. can you help me?

File: platformio/src/locales/locale_fr_FR.inc [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+                              "jeudi", "vendredi", "samedi"};
+// ABDAY_{1-7}             abbreviated ame of the n-th day of the week
+const char *LC_ABDAY[7]    = {"dim.", "lun.", "mar.", "mer.", "jeu.", "ven.", "sam."};
+// MON_{1-12}              name of the n-th month of the year
+const char *LC_MON[12]     = {"janvier", "f\xE9""vrier", "mars", "avril", 
+                               "mai",       "juin",     "juillet",     "ao\xFB""t", 
+                               "septembre", "octobre",  "novembre", "d\xE9""cembre"};
+// ABMON_{1-12}            abbreviated name of the n-th month of the year
+const char *LC_ABMON[12]   = {"janv.", "f\xE9""vr.", "mars", "avril", "mai", "juin", 
+                              "juil.", "ao\xFB""t", "sept.", "oct.", "nov.", ""d\xE9""c."};
```
Comment (lmarzen) [inline platformio/src/locales/locale_fr_FR.inc:56]: Whoops, I left an extra quotation mark in my last suggestion
```suggestion
const char *LC_ABMON[12]   = {"janv.", "f\xE9""vr.", "mars", "avril", "mai", "juin", 
                              "juil.", "ao\xFB""t", "sept.", "oct.", "nov.", "d\xE9""c."};
```

**Review by lmarzen — CHANGES_REQUESTED** [review]
One last thing

**Review by lmarzen — APPROVED** [review]
Looks good!

**Comment by lmarzen** [comment]
Thanks for you contribution.

**Pull request merged**